src/halmos/sevm.py

```diff
@@ -146,33 +180,61 @@ def byte_length(x: Any) -> int:
 
     raise ValueError(x)
 
+
 def instruction_length(opcode: Any) -> int:
     opcode = int_of(opcode)
     return (opcode - EVM.PUSH0 + 1) if EVM.PUSH1 <= opcode <= EVM.PUSH32 else 1
 
-def wextend(mem: List[Byte], loc: int, size: int) -> None:
-    if len(mem) < loc + size:
-        mem.extend([BitVecVal(0, 8) for _ in range(loc + size - len(mem))])
 
-def wload(mem: List[Byte], loc: int, size: int) -> Bytes:
+def wextend(mem: List[UnionType[int, BitVecRef]], loc: int, size: int) -> None:
+    mem.extend([0] * (loc + size - len(mem)))
+
+
+def wload(mem: List[UnionType[int, BitVecRef]], loc: int, size: int, prefer_concrete=False) -> UnionType[bytes,Bytes]:
     wextend(mem, loc, size)
 
+    memslice = mem[loc:loc+size]
+
+    # runtime sanity check: mem should only contain ints or BitVecs (not bytes)
+    if not all(isinstance(i, int) or is_bv(i) for i in memslice):
+        raise ValueError(memslice)
```

Comment:
would it be possible to incorporate this check into the existing iteration below instead of having a separate scan?  or would it be too complicated, and performance gain wouldn't justify the effort?